### PR TITLE
Handle server error when coming back to albums from album photos

### PR DESCRIPTION
**Pull request by CPatchane**

### CODE REVIEW

File: src/containers/AlbumsView.jsx [hunk 1]
```diff
@@ -15,13 +15,20 @@ export class AlbumsView extends Component {
...
+    this.handleErrorFromChildren = this.handleErrorFromChildren.bind(this)
+
     props.fetchAlbums()
       .then(() => { this.setState({isFetching: false, error: false}) })
       .catch(albumsError => {
         this.setState({isFetching: false, error: albumsError})
       })
   }
 
+  handleErrorFromChildren (error) {
```
Comment (gregorylegarec) [inline src/containers/AlbumsView.jsx]: I am not a big fan of the naming here. I think that we should be able to handle on error whatever its source is.

Si in my opinion, this method should be called `onError` and pass to children component via an attribute with the same name. It feels more idiomatic. What do you think ? What's @goldoraf opinion ?

File: src/containers/AlbumsView.jsx [hunk 2]
```diff
@@ -34,7 +41,10 @@ export class AlbumsView extends Component {
       !this.props.children
         ? <div>
           <Topbar viewName='albums' />
-          <AlbumsList {...this.props} {...this.state} />
+          <AlbumsList
+            {...this.props}
+            handleErrorToParent={this.handleErrorFromChildren}
```
Comment (gregorylegarec) [inline src/containers/AlbumsView.jsx]: As mentioned above, I think this attribute should be named `onError` (cc @goldoraf).

File: src/components/AlbumItem.jsx [hunk 3]
```diff
@@ -35,6 +35,9 @@ export class AlbumItem extends Component {
         url: link,
         isLoading: false
       }))
+      .catch(linkError => {
```
Comment (gregorylegarec) [inline src/components/AlbumItem.jsx]: To improve purity of this method, I think we may move the catch directly in the constructor, after `fetchPhoto` call. What do you think ?

File: src/containers/AlbumsView.jsx (hunk 1 above)
Comment (CPatchane) [inline src/containers/AlbumsView.jsx]: I am open about this point, I am not a big of the current naming as well, I just suggested here.

File: src/components/AlbumItem.jsx (hunk 3 above)
Comment (CPatchane) [inline src/components/AlbumItem.jsx]: Even if we keep the `then()` here?

File: src/containers/AlbumsView.jsx (hunk 1 above)
Comment (goldoraf) [inline src/containers/AlbumsView.jsx]: I agree with @gregorylegarec: `onError` sounds better here.

Comment (CPatchane) [inline src/containers/AlbumsView.jsx]: You're right, thanks, I'll fix it.

File: src/components/AlbumItem.jsx (hunk 3 above)
Comment (gregorylegarec) [inline src/components/AlbumItem.jsx]: I thank that `fetchPhoto` was returning a Promise but it is not the case. At this point, it is almost just a wrapper for `getPhotoLink`.  As I think that the real purpose of this method is to fetch the main photo of the album, what do you think about put the related logic inside it, like this:

```javascript
fetchMainPhoto(album) {
  if (album.photoIds && album.photosIds.length) {
       return getPhotoLink(album.photoIds[0])
  } else {
       return Promise.resolve(null)
  }
}
```
And call it like this :

```javascript
fetchMainPhoto(props.album || {})
  .then(link => this.setState({
    url: link,
    isLoading: false
  })
  .catch(linkError => {
    this.props.onServerError(linkError)
    this.setState({
      url: '',
      isLoading: false
    })
  })
```
Bonus : you don't need to bind this method.

**Comment by gregorylegarec** [comment]
Thanks @CPatchane !

**Pull request merged**